src/backend/db/database_manager.py

```diff
@@ -1813,4 +1813,262 @@ def get_paginated_files(self, customer_guid: str, page: int = 1, page_size: int
+            FROM common_db.customer_file_status 
+            WHERE to_be_deleted = TRUE
+            ORDER BY delete_request_timestamp ASC
+            LIMIT :num_of_records
+            """
+            return session.execute(text(query), {"num_of_records": num_of_records}).fetchall()
+        except SQLAlchemyError as e:
+            logger.error(f"Error fetching pending deletions: {e}")
+            return []
+        finally:
+            session.close()
+
+    def update_deletion_status(self, customer_guid, filename, new_status, error, error_retry):
+        """Mirrors update_status exactly but for deletion status fields"""
+        session = self._session_factory()
+        try:
+            # Update common_db.customer_file_status
+            update_common_query = """
+                UPDATE common_db.customer_file_status
+                SET delete_status = :new_status,
+                errors = :error,
+                error_retry = :error_retry,
+                current_activity_updated_time = CURRENT_TIMESTAMP(6)
+                WHERE customer_guid = :customer_guid AND filename = :filename
+            """
+            session.execute(text(update_common_query), {
+                'new_status': new_status,
+                'error': error,
+                'error_retry': error_retry,
+                'customer_guid': customer_guid,
+                'filename': filename
+            })
+
+            # Update customer-specific uploadedfile_status
+            customer_db = self.get_customer_db(customer_guid)
+            update_customer_query = f"""
+                UPDATE `{customer_db}`.uploadedfile_status
+                SET delete_status = :new_status,
+                errors = :error,
+                error_retry = :error_retry,
+                current_activity_updated_time = CURRENT_TIMESTAMP(6)
+                WHERE customer_guid = :customer_guid AND filename = :filename
+            """
+            session.execute(text(update_customer_query), {
+                'new_status': new_status,
+                'error': error,
+                'error_retry': error_retry,
+                'customer_guid': customer_guid,
+                'filename': filename
+            })
+
+            session.commit()
+            logger.info(
+                f"Updated deletion status for {filename} to {new_status} "
+                f"with error: {error} for customer: {customer_guid}"
+            )
+        except SQLAlchemyError as e:
+            logger.error(f"Error updating deletion status for {filename}: {e}")
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def finalize_deletion(self, customer_guid: str, filename: str, error):
+        session = self._session_factory()
+        try:
+            customer_db = self.get_customer_db(customer_guid)
+            if error:
+                # Mark as error in both tables
+                update_common_query = """
```

Comment:
Since this is the final error
We can keep the data customer_customer_guid database and remove the row from common db

```
                update_common_query = """
                    UPDATE common_db.customer_file_status
                    SET delete_status = 'error'
                    WHERE customer_guid = :customer_guid AND filename = :filename
                """
                session.execute(text(update_common_query), {
                    'customer_guid': customer_guid,
                    'filename': filename
                })
```